viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
+
+#create set of titles that friends have watched
+def friend_watched_set(user_data):
+        friend_watched_titles_list = []
+        for friends_watched_dict in user_data["friends"]:
+            for friends_watched_movies in friends_watched_dict["watched"]:
+                friend_watched_titles_list.append(friends_watched_movies['title'])
+        friend_watched_titles_set = set(friend_watched_titles_list)
+        return friend_watched_titles_set
+
+#create set of titles the user has watched
+def user_watched_set(user_data):
+    user_watched_titles_list = []
+    for user_watched_movie in user_data["watched"]:
+       user_watched_titles_list.append(user_watched_movie['title'])
+    user_watched_titles_set = set(user_watched_titles_list)
+    return user_watched_titles_set
+
+def get_unique_watched(user_data):
+    #find the difference between the sets
+    movie_titles_unique_to_user = user_watched_set(user_data).difference(friend_watched_set(user_data))
+
+    #iterate through the user_watched list and append the movie dictionary to a list if the movie's title is in the unique to user set.
+    unique_to_user_watched = []
+    for movie in user_data["watched"]:
+        if movie["title"] in movie_titles_unique_to_user:
+            unique_to_user_watched.append(movie)
+
+    return unique_to_user_watched
+
+def get_friends_unique_watched(user_data):
+    #find the movies unique to friends
+    movie_titles_unique_to_friends = friend_watched_set(user_data).difference(user_watched_set(user_data))
+
+
+    #iterate through friends movies and append the movie dictionary to a list if the movie's title is in movie_titles_unique_to friends set
+    unique_to_friends_watched = []
+
+    for friends_watched_dict in user_data["friends"]:
+            for friends_watched_movie in friends_watched_dict["watched"]:
+                if friends_watched_movie["title"] in movie_titles_unique_to_friends:
+                    if friends_watched_movie in unique_to_friends_watched:
+                        continue
+                    else:
+                        unique_to_friends_watched.append(friends_watched_movie)
+    
+    return unique_to_friends_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    recommended_movies = []
+
+    for friend in user_data["friends"]:
+        for movie_dict in friend["watched"]:
+            if movie_dict["host"] not in user_data["subscriptions"]:
+                continue
+            elif movie_dict not in user_data["watched"]:
+                recommended_movies.append(movie_dict)
+
+    return recommended_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    favorite_genre = get_most_watched_genre(user_data)

```

Comment:
Great! This helper function is very applicable and helps to DRY (don't-repeat-yourself) your code.